pkg/ingester/ingester.go

```diff
@@ -2954,3 +3014,19 @@ func (i *Ingester) UserRegistryHandler(w http.ResponseWriter, r *http.Request) {
 		Timeout:            10 * time.Second,
 	}).ServeHTTP(w, r)
 }
+
+// checkReadOverloaded checks whether the ingester read path is overloaded wrt. CPU and/or memory.
+func (i *Ingester) checkReadOverloaded() error {
+	if !i.cfg.UtilizationLimitsEnabled {
+		return nil
+	}
+
+	reason := i.utilizationBasedLimiter.LimitingReason.Load()
+	if reason == "" {
+		return nil
+	}
+
+	level.Debug(i.logger).Log("msg", "rejecting read request due to resource utilization based limiting",
+		"reason", reason)
```

Comment:
[nit] I would remove this debug log line. Debug is typically disabled in prod and so this doesn't add much value.

Reply:
Done.